internal/store/verticalpodautoscaler.go

```diff
@@ -0,0 +1,256 @@
+)
+
+var (
+	descVerticalPodAutoscalerLabelsName          = "kube_verticalpodautoscaler_labels"
+	descVerticalPodAutoscalerLabelsHelp          = "Kubernetes labels converted to Prometheus labels."
+	descVerticalPodAutoscalerLabelsDefaultLabels = []string{"namespace", "verticalpodautoscaler", "target_api_version", "target_kind", "target_name"}
+
+	vpaMetricFamilies = []metric.FamilyGenerator{
+		{
+			Name: descVerticalPodAutoscalerLabelsName,
+			Type: metric.Gauge,
+			Help: descVerticalPodAutoscalerLabelsHelp,
+			GenerateFunc: wrapVPAFunc(func(a *autoscaling.VerticalPodAutoscaler) *metric.Family {
+				labelKeys, labelValues := kubeLabelsToPrometheusLabels(a.Labels)
+				return &metric.Family{
+					Metrics: []*metric.Metric{
+						{
+							LabelKeys:   labelKeys,
+							LabelValues: labelValues,
+							Value:       1,
+						},
+					},
+				}
+			}),
+		},
+		{
+			Name: "kube_verticalpodautoscaler_update_mode",
+			Type: metric.Gauge,
+			Help: "Update mode of the VerticalPodAutoscaler.",
+			GenerateFunc: wrapVPAFunc(func(a *autoscaling.VerticalPodAutoscaler) *metric.Family {
+				ms := []*metric.Metric{}
+
+				if a.Spec.UpdatePolicy.UpdateMode == nil {
+					return &metric.Family{
+						Metrics: ms,
+					}
+				}
+
+				for _, mode := range []autoscaling.UpdateMode{
+					autoscaling.UpdateModeOff,
+					autoscaling.UpdateModeInitial,
+					autoscaling.UpdateModeRecreate,
+					autoscaling.UpdateModeAuto,
+				} {
+					var v float64
+					if *a.Spec.UpdatePolicy.UpdateMode == mode {
+						v = 1
+					} else {
+						v = 0
+					}
+					ms = append(ms, &metric.Metric{
+						LabelKeys:   []string{"update_mode"},
+						LabelValues: []string{string(mode)},
+						Value:       v,
+					})
+				}
+
+				return &metric.Family{
+					Metrics: ms,
+				}
+			}),
+		},
+		{
+			Name: "kube_verticalpodautoscaler_container_resource_policy_min",
+			Type: metric.Gauge,
+			Help: "Minimum resources the VerticalPodAutoscaler can set for containers matching the name.",
+			GenerateFunc: wrapVPAFunc(func(a *autoscaling.VerticalPodAutoscaler) *metric.Family {
+				ms := []*metric.Metric{}
+				for _, c := range a.Spec.ResourcePolicy.ContainerPolicies {
+					ms = append(ms, vpaResourcesToMetrics(c.ContainerName, c.MinAllowed)...)
+
+				}
+				return &metric.Family{
+					Metrics: ms,
+				}
+			}),
+		},
+		{
+			Name: "kube_verticalpodautoscaler_container_resource_policy_max",
+			Type: metric.Gauge,
+			Help: "Maximum resources the VerticalPodAutoscaler can set for containers matching the name.",
+			GenerateFunc: wrapVPAFunc(func(a *autoscaling.VerticalPodAutoscaler) *metric.Family {
+				ms := []*metric.Metric{}
+				for _, c := range a.Spec.ResourcePolicy.ContainerPolicies {
+					ms = append(ms, vpaResourcesToMetrics(c.ContainerName, c.MaxAllowed)...)
+				}
+				return &metric.Family{
+					Metrics: ms,
+				}
+			}),
+		},
+		{
+			Name: "kube_verticalpodautoscaler_container_status_recommendation_lower_bound",
```

Comment:
naming same scheme as `kube_verticalpodautoscaler_container_resource_policy_min`, although unfortunately due to the API definition, this is gonna end up a bit weird, but I'd still prefer to stick to systematic naming, so something like: `kube_verticalpodautoscaler_status_recommendation_containerrecommendations_lower_bound`